docs(linter): Mark a dozen more rules as unsupported.

connorshea opened the pull request:
This backports most of the rule addition/removal changes from #17512.

- ~~I removed [`vue/define-macros-order`](https://eslint.vuejs.org/rules/define-macros-order.html#vue-define-macros-order) from the unsupported list, as it seems like it does not require template parsing and should be possible to implement?~~
- Added `n/prefer-node-protocol`, `n/no-process-exit`, and `n/file-extension-in-import` to the unsupported rule list, as they are implemented by other rules in oxlint already.
- Added `vue/no-unused-properties` and `vue/no-undef-properties` as they both require Vue template parsing.
- Added `jest/no-unnecessary-assertion`, `jest/unbound-method`, `jest/no-error-equal`, `vitest/unbound-method`, and `vitest/prefer-vi-mocked` to unsupported rules with rationale that they require type information.

~~I also added [`vue/jsx-uses-vars`](https://eslint.vuejs.org/rules/jsx-uses-vars.html#vue-jsx-uses-vars) as an unsupported rule with the rationale that it _should_ already be implemented via no-unused-vars, as it only applies to vue components using JSX. Though I'm not 100% sure if that is actually supported properly at present for Vue files.~~

connorshea commented:
Found the source, you are correct on the origin of this rule being excluded: https://github.com/oxc-project/oxc/pull/11442#pullrequestreview-2890081664

Sysix commented:
Happy with both version (unsupported/supported). Let see what the oxlint owner wants :P

connorshea committed:
docs(linter): Mark a dozen more rules as unsupported. (#17514)

This backports most of the rule addition/removal changes from #17512.

- ~~I removed [`vue/define-macros-order`](https://eslint.vuejs.org/rules/define-macros-order.html#vue-define-macros-order) from the unsupported list, as it seems like it does not require template parsing and should be possible to implement?~~
- Added `n/prefer-node-protocol`, `n/no-process-exit`, and `n/file-extension-in-import` to the unsupported rule list, as they are implemented by other rules in oxlint already.
- Added `vue/no-unused-properties` and `vue/no-undef-properties` as they both require Vue template parsing.
- Added `jest/no-unnecessary-assertion`, `jest/unbound-method`, `jest/no-error-equal`, `vitest/unbound-method`, and `vitest/prefer-vi-mocked` to unsupported rules with rationale that they require type information.

~~I also added [`vue/jsx-uses-vars`](https://eslint.vuejs.org/rules/jsx-uses-vars.html#vue-jsx-uses-vars) as an unsupported rule with the rationale that it _should_ already be implemented via no-unused-vars, as it only applies to vue components using JSX. Though I'm not 100% sure if that is actually supported properly at present for Vue files.~~

camc314 reviewed (commented):
lgtm ✅
